tests/test_factorgraph.py

```diff
@@ -1381,3 +1381,105 @@ def test_clear_beliefs():
     assert bp.get_belief_from_var("x", "s1") is not None
     assert bp.get_belief_from_var("a", "s1") is not None
     assert bp.get_belief_from_var("b", "s1") is not None
+
+
+def test_single_gf_sanity_check():
+    from scalib.attacks.factor_graph import GenFactor
+
+    nc = 13
+    graph = f"""NC {nc}
+    VAR MULTI a
+    VAR MULTI b
+    VAR MULTI c
+    VAR MULTI d
```

Comment:
There should be a more varied mix of SINGLE/MULTI vars and factors in the tests.